src/core/mac/mac.cpp

```diff
@@ -569,6 +596,12 @@ void Mac::ScheduleNextTransmission(void)
     else
     {
         mState = kStateIdle;
+
+        if (mDisableRadioOnIdle)
+        {
+            mDisableRadioOnIdle = false;
+            DisableRadio();
+        }
```

Comment:
Probably not a practical use case, however a simple ref count solution would solve it.